### PR TITLE
Modal dialog service

### LINKED ISSUE #98: Using page.showModal with @Component's
Given `my.component` is an angular `Component`:
```
page.showModal('./pages/my.component', 'Modal Context', () => {
 // close function
}, true);
```

Wasn't too surprised to see it throw this:
```
file:///app/tns_modules/ui/frame/frame-common.js:83:94: JS ERROR Error: Failed to load Page from entry.moduleName: ./pages/my.component
```

Curious how to integrate the page's `showModal` api with `nativescript-angular`.

### PR DESCRIPTION
**Pull request by vakrilov**
Resolves #98

### CODE REVIEW

**Comment by NathanWalker** [comment]
woo-hoo! Thanks @vakrilov !

**Comment by anhoev** [comment]
This is cool . Does it support custom template or only text ?

**Comment by vakrilov** [comment]
@anhoev You pass a component type to the `showModal` method and it loads it in a dialog - [like that](https://github.com/NativeScript/nativescript-angular/blob/modal-pages/ng-sample/app/examples/modal/modal-test.ts#L40-L42).

**Comment by anhoev** [comment]
@vakrilov :+1: 
Super !

File: src/nativescript-angular/directives/dialogs.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            ]);
+
+            this.loader.loadNextToLocation(DetachedLoader, this.elementRef, providers)
+                .then((loaderRef) => {
+                    detachedLoaderRef = loaderRef;
+                    return (<DetachedLoader>loaderRef.instance).loadComponent(type)
+                })
+                .then((compRef) => {
+                    //Component loaded. Find its root native view.
+                    const componenetView = <View>compRef.location.nativeElement;
```
Comment (hdeshev) [inline src/nativescript-angular/directives/dialogs.ts]: Typo: componentView

File: tests/app/tests/modal-dialog.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import {assert} from "./test-config";
+import {Component, ComponentRef} from "angular2/core";
+import {TestApp} from "./test-app";
+import {Page} from "ui/page";
+import {topmost} from "ui/frame";
+import {ModalDialogHost, ModalDialogOptions, ModalDialogParams, ModalDialogService} from "../nativescript-angular/directives/dialogs";
+
+@Component({
+    selector: "modal-comp",
+    template: `<Label text="this is modal componenet"></Label>`
```
Comment (hdeshev) [inline tests/app/tests/modal-dialog.ts]: Typo: component

File: tests/app/tests/modal-dialog.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    }
+}
+
+@Component({
+    selector: "fail-comp",
+    providers: [ModalDialogService],
+    template: `<Label text="This app is doomed"></Label>`
+
+})
+export class FailComponenet {
```
Comment (hdeshev) [inline tests/app/tests/modal-dialog.ts]: Typo: component

File: tests/app/tests/modal-dialog.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+@Component({
+    selector: "sucess-comp",
+    directives: [ModalDialogHost],
+    providers: [ModalDialogService],
+    template: `
+    <GridLayout modal-dialog-host margin="20">
+        <Label text="Modal dialogs"></Label>
+    </GridLayout>`
+})
+export class SuccessComponenet {
```
Comment (hdeshev) [inline tests/app/tests/modal-dialog.ts]: Typo: component

File: tests/app/tests/modal-dialog.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    });
+
+    afterEach(() => {
+        var page = topmost().currentPage;
+        if (page && page.modal) {
+            console.log("Warning: closing a leftover modal page!");
+            page.modal.closeModal();
+
+        }
+        testApp.disposeComponenets();
```
Comment (hdeshev) [inline tests/app/tests/modal-dialog.ts]: Typo: disposeComponents

**Comment by hdeshev** [comment]
:100:

**Pull request merged**